### DISCUSSION TITLE
Scrape album uris/urls from song and artist searches

### DISCUSSION THREAD

**Discussion by shillshocked** [Aran404/SpotAPI#7] [Ideas]
I've found your API to be more capable than Spotify's current API as they have limited the average user so much. But to make your API more capable, it would be very useful if we could obtain the album uris or urls from the song and artist searches. Since there is no album search implemented, this would be an interim solution.

**Comment by Aran404** [comment]
> I've found your API to be more capable than Spotify's current API as they have limited the average user so much. But to make your API more capable, it would be very useful if we could obtain the album uris or urls from the song and artist searches. Since there is no album search implemented, this would be an interim solution.

You can probably construct that information through the Song API. IIRC there's a lot of meta data in the response so you could most likely do that with ease. If I'm wrong please let me know, I haven't looked at this code in 3 months.

**Comment by shillshocked** [comment]
I can get track, album, and artist info through this API. But I cannot get artist albums, which would be a real timesaver. It would be nice if that part of the Spotify API could be emulated somehow (scraping an artist page's releases?)

**Comment by Aran404** [comment]
I'll see what I can do, feel free to open a pull request if you can figure it out

**Comment by shillshocked** [comment]
Some limited support for artist_id has been added. But it returns a finite number of results when you use it.

It looks like the persisted query that SpotAPI uses for “queryArtistOverview” doesn’t actually support adding extra pagination parameters (like “albumsLimit” or “albumsOffset”) in its variables. In a test script I tried to inject those keys into the JSON sent to that endpoint, but the endpoint ignores them – or worse, returns no album items at all.

The original SpotAPI script simply calls get_artist(artist_id) and then extracts the albums from the nested data under
  data → artistUnion → discography → albums → items
and that “items” list is returned without any pagination support. (In other words, the backend query already returns only a fixed subset of the artist’s releases.)

To properly paginate artist albums using SpotAPI we need to extend or change SpotAPI so that it uses a different persisted query (if one exists) or manually makes calls that support pagination. However, as it stands, the “queryArtistOverview” operation does not accept additional paging parameters.

So using the Spotify Web API, for example, I can get 10x more releases than spotapi because it paginates the artists when there are more than "x" number of releases in a query.

Any ideas about how to address this?

**Reply by Aran404** [reply]
I'd appreciate the non chatgpt response, but I'll look into it

**Reply by shillshocked** [reply]
Well, I have been using AI to help me code my scripts as I hit a bit of a wall over the past few years. It was a long-term project. This API has been a nice addition to my arsenal. :)
